info-compiler/src/main/java/com/google/sps/webcrawler/NewsContentProcessor.java

```diff
@@ -87,46 +85,54 @@ public int compare(PageRank.Result a, PageRank.Result b) {
   };
 
   /** Extracts the first {@code MAX_WORD_COUNT} words from the news article content. */
-  public static NewsArticle abbreviate(NewsArticle originalNewsArticle) {
-    NewsArticle newsArticle = new NewsArticle(originalNewsArticle);
+  public static void abbreviate(NewsArticle newsArticle) {
     String[] splitContent = newsArticle.getContent().split(" ");
     int wordCount = splitContent.length;
     int allowedLength = Math.min(wordCount, MAX_WORD_COUNT);
     String abbreviatedContent =
         String.join(" ", Arrays.asList(splitContent).subList(0, allowedLength));
     newsArticle.setAbbreviatedContent(abbreviatedContent);
-    return newsArticle;
   }
 
   /**
    * Extractively summarizes the news article content by computing inter-sentence similarity and
    * applying PageRank to find the most meaningful sentences. The chosen sentences are arranged
-   * in the original order they appeared in.
+   * in the original order they appeared in. Sets empty summarized content in the event of an
+   * exception, such as if {@code SentenceModel} or {@code TokenizerModel} model instantiation
+   * fails, of if {@code PageRank} algorithm fails. {@code SENTENCE_DETECTOR_FILE} and {@code
+   * TOKENIZER_FILE} must be prepared.
    *
-   * @throws Exception if {@code SentenceModel} or {@code TokenizerModel} model instantiation
-   *     fails, of if {@code PageRank} algorithm fails. For example, {@code SENTENCE_DETECTOR_FILE}
-   *     and {@code TOKENIZER_FILE} must be prepared.
    * @see <a href="https://towardsdatascience.com/understand-text-summarization-and-create-your-
    *     own-summarizer-in-python-b26a9f09fc70></a>
    */
-  public static NewsArticle summarize(NewsArticle originalNewsArticle) throws Exception {
-    NewsArticle newsArticle = new NewsArticle(originalNewsArticle);
+  public static void summarize(NewsArticle newsArticle) {
     String rawContent = newsArticle.getContent();
-    String[] sentences = breakIntoSentences(rawContent);
+    String[] sentences;
+    try {
+      sentences = breakIntoSentences(rawContent);
+    } catch (IOException e) {
+      newsArticle.setSummarizedContent("");
+      return;
+    }
     if (sentences.length <= SUMMARIZATION_MAX_SENTENCE_NUMBER) {
       newsArticle.setSummarizedContent(rawContent);
-      return newsArticle;
+      return;
+    }
+    List<PageRank.Result<IntValue>> ranking;
+    try {
+      Graph<IntValue, StringValue, DoubleValue> similarityGraph = buildSimilarityGraph(sentences);
+      ranking = getRanking(similarityGraph);
+    } catch (Exception e) {
+      newsArticle.setSummarizedContent("");
+      return;
     }
```

Comment:
Would it make sense to combine these try blocks? You can still return things (i.e. if sentences < max #) from within the try block.